src/plugins/telemetry/prometheus/prometheus_exporter.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "prometheus_exporter.h"
+#include "common/nixl_log.h"
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <iomanip>
+#include <thread>
+#include <chrono>
+
+namespace {
+const uint16_t prometheusExporterDefaultPort = 9090;
+
+const char prometheusPortVar[] = "NIXL_TELEMETRY_PROMETHEUS_PORT";
+const char prometheusLocalVar[] = "NIXL_TELEMETRY_PROMETHEUS_LOCAL";
+
+const std::string prometheusExporterTransferCategory = "NIXL_TELEMETRY_TRANSFER";
+const std::string prometheusExporterPerformanceCategory = "NIXL_TELEMETRY_PERFORMANCE";
+const std::string prometheusExporterMemoryCategory = "NIXL_TELEMETRY_MEMORY";
+const std::string prometheusExporterBackendCategory = "NIXL_TELEMETRY_BACKEND";
+const std::string prometheusExporterLocalAddress = "127.0.0.1";
+const std::string prometheusExporterPublicAddress = "0.0.0.0";
+
+static uint16_t
+getPort() {
+    auto port_str = std::getenv(prometheusPortVar);
+    if (!port_str) {
+        return prometheusExporterDefaultPort;
+    }
+
+    try {
+        int port = std::stoi(port_str);
+        if (port < 1 || port > 65535) {
+            NIXL_WARN << "Invalid port number " << port
+                      << ", must be between 1-65535. Using default: "
+                      << prometheusExporterDefaultPort;
+            port = prometheusExporterDefaultPort;
+        }
+        return port;
+    }
+    catch (const std::exception &e) {
+        NIXL_WARN << "Invalid port " << port_str
+                  << "', expected numeric port. Using default: " << prometheusExporterDefaultPort;
+        return prometheusExporterDefaultPort;
+    }
```

Comment:
Optional change. You can skip it if you think the current version is better.
```suggestion
    try {
        int port = std::stoi(port_str);
        if (port < 1 || port > 65535) {
            throw std::out_of_range;
        }
        return port;
    }
    catch (const std::exception &e) {
        NIXL_WARN << "Invalid port '" << port_str
                  << "', expected numeric port between 1-65535. Using default: "
                  << prometheusExporterDefaultPort;
        return prometheusExporterDefaultPort;
    }
```